addons/pos_loyalty/static/src/overrides/models/pos_store.js

```diff
@@ -582,9 +582,15 @@ patch(PosStore.prototype, {
     },
 
     computeDiscountProductIdsForAllRewards(data) {
-        const products = this.models[data.model].readMany(data.ids);
+        const productsSerialized = this.models[data.model].readMany(data.ids).map((p) => {
+            return {
+                product: p,
+                serialized: p.serialize(),
+            };
+        });
+
         for (const reward of this.models["loyalty.reward"].getAll()) {
-            this.compute_discount_product_ids(reward, products);
+            this.compute_discount_product_ids(reward, undefined, productsSerialized);
```

Comment:
```suggestion
            this.compute_discount_product_ids(reward, products, productsSerialized);
```

Continue to give products to this method; in the case of customization, this may be useful.